src/System.CommandLine/Help/HelpBuilder.cs

```diff
@@ -526,8 +526,8 @@ private IEnumerable<HelpItem> GetArgumentHelpItems(ISymbol symbol)
 
         protected virtual string ArgumentDescriptor(IArgument argument)
         {
-            if (argument.ValueType == typeof(bool) ||
-                argument.ValueType == typeof(bool?))
+            if (argument.Parents.OfType<IOption>().Any()
```

Comment:
Is the parent check needed here?

Reply:
That's the part I'm least clear on. From my looking through history on this line, tracking back to #495 and #500, it appears that the intention is to not show the suggestions on boolean _options_, but that means that an `Argument` on a command only outputs the description instead of suggesting an input.

I'm not 100% sure this was the correct check, but I also didn't see a better one. From:
```cs
var rootCommand = new RootCommand
 {
    new Argument<bool>("value1", "Some value"),
    new Argument<int>("value2", "Another value"),
};
```

### Previous
```
Usage:
  CommandLineReproducer [options] [<value1> <value2>]

Arguments:
              Some value
  <value2>    Another value

Options:
  --version         Show version information
  -?, -h, --help    Show help and usage information
```

### Proposed
```
Usage:
  CommandLineReproducer [options] [<value1> <value2>]

Arguments:
  <False|True>   Some value
  <value2>       Another value

Options:
  --version         Show version information
  -?, -h, --help    Show help and usage information
```

Reply:
Looking at this now, I'm not sure special-casing `bool`/`bool?` is correct at all.

So what should we show?

The proposal here isn't consistent with how we show allowed values elsewhere, though how we do that is also not ideal. 

For comparison, here's comparable output from System.CommandLine (`t-rex`) and `git`, with allowed values circled:

![image](https://user-images.githubusercontent.com/547415/105105081-3d829a00-5a68-11eb-95bb-38f008d81ba7.png)

I wonder if we should do something more like the `t-rex` example for `bool` as well, e.g.:

```console
Usage:
  CommandLineReproducer [options] [<value1> <value2>]

Arguments:
  value1 <False|True>   Some value
  value2                         Another value

Options:
  --version         Show version information
  -?, -h, --help    Show help and usage information
```

Reply:
I agree that is better, although to some extent it's reverting #495. I'd picture this landing as:

1. Suppress the `<true|false>` on an `Option` when the default is false. (This is #495)
2. Show `<true|false>` on Options when the default is true. (Not sure it does this today)
3. Always show `<true|false>` on arguments that are separate from an option. (Resolving #1157)

Reply:
At the time we fixed #495, the display of default values hadn't been implemented. I do think it's awkward that we have the allowable values in the left column and the default value in the right column. 

If we combined the allowed and default values into a single piece of the help, could it be made clearer overall?

Reply:
Maybe, but I'm not entirely sure what you mean that to end up being.

Reply:
I'm not sure. Open to suggestions.

Reply:
Ah - I'm not sure what the state of this PR is but having the documentation like this is wrong:

```
Arguments:
  <False|True>   Some value
  <value2>       Another value
```

It should be:

```
Arguments:
  <value1>       Some value
  <value2>       Another value
```

Reply:
@skrysmanski The state at the moment is to take the time to evaluate help output as a whole in light of your issue and a couple others, #984 and #987. I agree that the current implementation is wrong, and I'm open to suggestions. What do you think about this:
```
Arguments:
  <value1>                 Some value
    allowed: true|false
  <value2>                 Another value
  <value3>                 An enum value
    allowed: Read|ReadWrite|Write
```

Reply:
@adamhewitt627 I have no personal preference on either option. My only point was that the name of the argument was missing. I know how complicated it can be to have proper help pages. This is why I'm using `System.CommandLine` so that I don't need to think about this ;)

Reply:
Since this discussion seems to be larger than the current issue, I opened #1174 to discuss.